### ISSUE TITLE
Certificate error in v1.21.0

### ISSUE THREAD

**Issue by LSmyrnaios** [kubernetes/kompose#1251]
Hi, I tried to do a `kompose up` and I got the following error:

> FATA Error while deploying application: Get https://127.0.0.1:6443/api: x509: certificate is valid for 10.96.0.1, <PC's IP>, not 127.0.0.1 

I solved it, by setting the new option `--server https://<PC's IP>:6443`

I would like to ask:
a) Why did I received this error about the certificate?
b) Why `kompose` expected everything to run smoothly on `127.0.0.1`? Is there a default configuration that's changed on my machine?
b) Why `kompose` cannot do an auto-detect and manage to run in any IP-case?

Versions:
Kubernetes v1.15.10
Kompose: v1.21.0

**Comment by hangyan** [comment]
1. The cert for your cluster contains info about valid ip it support ,it dose not contains 127.0.0.1
2/3. kompose is a very old project, the `client-go` package it rely on it outdated and does not support many new features about  reading server ip from kubeconfig, so it use the default address `https://127.0.0.1:6443`,

I know the current solution is not perfect, but it we want to upgrade it, it requires a lot of refactor of the current code. So that's the situation for now. It works fine for most users, and if it's not, the `--server` flag can do some help

**Comment by LSmyrnaios** [comment]
Thanks for your response.

1. Will it be solved if I allow the "loopback address" in the firewall or you think there's something else going on?
2. & 3. I see, "client-go" and "kubernetes" dependencies are outdated for more than 3 years.
Sooner or later, kompose will need this upgrade to support new kubernetes stuff.
Also, the api used [here](https://github.com/kubernetes/kompose/tree/master/vendor/k8s.io/kubernetes/federation/apis/federation) will get unsupported at some point.
I guess you could open a new issue to welcome and coordinate people in upgrading those dependencies in the future.

**Comment by hangyan** [comment]
1. The allowed ip address is written in the cert file...
2. Yes, we are aware of this issue. I have tried a few times to solve this issue. It basically need to  rewrite 70% of the codebase, it's a huge task.  Nowadays, most people are familiar with Kubernetes already, they do not need this kind of tools anymore. The project has finished it's purpose as bridge from compose to kubernetes. You can check out the https://github.com/docker/compose-on-kubernetes project, it's more up-to-date with latest kubernetes versions.

**Comment by hangyan** [comment]
closed by #1305

**Comment by RichardBronosky** [comment]
Once you get the IP correct, you are very likely to get a `certificate signed by unknown authority` error. For that, you have to get your workstation's OS to accept the cert. For Linux, I use:

```
openssl s_client -showcerts -connect 127.0.0.1:6443 2>/dev/null </dev/null | \
  sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p' | \
  sudo tee /usr/local/share/ca-certificates/k8s.crt
sudo update-ca-certificates
sudo systemctl restart docker
```
